Fix crash when unable to serialize test parameter (#448)

TonEnfer opened the pull request:
<!---
Thank you so much for sending us a pull request! 

Make sure you have a clear name for your pull request. 
The name should start with a capital letter and no dot is required in the end of the sentence.
To link the request with issues use the following notation: (fixes #123, fixes #321\)

An example of good pull request names:
* Add Cucumber integration (fixes #123\)
* Add an ability to disable default plugins
* Support emoji in test descriptions
-->

### Context
<!---
Describe the problem or feature in addition to a link to the issues
-->
Fix #448 
#### Checklist
- [x] [Sign Allure CLA][cla]
- [ ] Provide unit tests

[cla]: https://cla-assistant.io/accept/allure-framework/allure2

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

The serializer skips fields that contain loop references and fields that could not be serialized

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

Re-format code

delatrie commented on Allure.Net.Commons/Functions/FormatFunctions.cs:
```diff
@@ -37,6 +43,23 @@ public static string Format(object? value)
+        }
+        catch
+        {
+            return JsonConvert.Undefined;
```
Let's fall back to `ToString` here. At least we'll get different values for different types (or even something useful if this method is overridden).
```suggestion
            return value.ToString();
```

TonEnfer committed:
Update Allure.Net.Commons/Functions/FormatFunctions.cs

Co-authored-by: Maxim <17935127+delatrie@users.noreply.github.com>

TonEnfer committed:
Update FormatFunctions.cs

delatrie reviewed (changes requested):
Let's remove indentation. It doesn't play well with Allure parameters.

TonEnfer committed:
Fixed a bug that caused a test to crash when there was a test parameter serialization error #448

Fixed formatting during serialization
Serialization parameters are moved to a private static field to improve performance
